sklearn/gaussian_process/kernels.py

```diff
@@ -1787,3 +1787,182 @@ def is_stationary(self):
     def __repr__(self):
         return "{0}(gamma={1}, metric={2})".format(
             self.__class__.__name__, self.gamma, self.metric)
+
+
+class SelectDimensionKernel(Kernel):
+    """Wrapper for applying kernels on selected active dimensions of input data.
+
+    Parameters
+    ----------
+    kernel : a kernel object.
+        an instantiation of kernel object.
+
+    active_dims: array-like, shape (n_active_dims,) or (n_features,)
+         If `active_dims` is a boolean array, then it should be of size
+         `n_features` marked as True for an active index.
+         If `active_dims` is an integer array, then it should be size of
+         `n_active_dims` denoting the indices of the active dimensions.
+
+    """
+
+    def __init__(self, kernel, active_dims):
+        self.active_dims = active_dims
+        self.kernel = kernel
+
+    def __call__(self, X, Y=None, eval_gradient=False):
+        """Return the kernel k(X, Y) and optionally its gradient on selected
+           dimensions.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples_X, n_features)
+            Left argument of the returned kernel k(X, Y)
+
+        Y : array, shape (n_samples_Y, n_features), (optional, default=None)
+            Right argument of the returned kernel k(X, Y). If None, k(X, X)
+            evaluated instead.
+
+        eval_gradient : bool (optional, default=False)
+            Determines whether the gradient with respect to the kernel
+            hyperparameter is determined. Only supported when Y is None.
+
+        Returns
+        -------
+        K : array, shape (n_samples_X, n_samples_Y)
+            Kernel k(X[:,active_dims], Y[:,active_dims])
+
+        K_gradient : array (opt.) shape (n_samples X n_samples X active_dims)
+            The gradient of the kernel k(X, X) with respect to the
+            hyperparameter of the kernel. Only returned when eval_gradient
+            is True.
+        """
+
+        active_dims = np.asarray(self.active_dims)
+
+        if (np.issubdtype(active_dims.dtype, np.bool) &
+           (active_dims.shape == X.shape[1])):
```

Comment:
what if active_dims is a boolean array but it does not have the correct shape? I think we should check this and throw a ValueError rather than converting it to int